packages/webapp/src/containers/Animals/Inventory/useAnimalInventory.tsx

```diff
@@ -214,6 +217,22 @@ interface BuildInventoryArgs {
   defaultAnimalTypes: DefaultAnimalType[];
 }
 
+const sortAnimalsIDs = (inventory: AnimalInventory[]) => {
+  return inventory.sort((a, b) => {
+    if (a.name && !b.name) {
+      return -1;
+    } else if (!a.name && b.name) {
+      return 1;
+    } else {
+      if (a.identification.length > b.identification.length) {
+        return 1;
+      } else {
+        return a.identification.localeCompare(b.identification);
+      }
```

Comment:
@Tbrid thank you so much for contributing!!

Unfortunately I don't think this portion of the sorting function is correct, and in testing it I'm not getting the right sorting order. On one hand, we shouldn't be using the `identification` property to sort since that's a string that can be either the name, the identifier or the internal identifier, but in this case we want to sort granularly, one property at the time. On the other hand, I'm not sure I understand the reasoning behind comparing the length of the strings?

I think this is more or less what the function could look like

```js
return inventory.sort((a, b) => {
    if (a.name && !b.name) {
      return -1;
    }
    if (b.name && !a.name) {
      return 1;
    }
    if (a.identifier && !b.identifier) {
      return -1;
    }
    if (b.identifier && !a.identifier) {
      return 1;
    }
    return (
      (a.name && b.name && a.name.localeCompare(b.name)) ||
      (a.identifier && b.identifier && a.identifier.localeCompare(b.identifier)) ||
      a.internal_identifier - b.internal_identifier
    );
  });
```

but I haven't tested it thoroughly!
If you have the time, a good idea could be to add some unit tests to verify that it's working properly. Otherwise we can just test it manually and make sure it works with a variety of records.

Reply:
I get that the length check was for the internal_identifier string that includes 'ID###' . 

If the `identification` property is no longer used here (it is just for label display now) then it should be a plain number as Anto described 👍